users/ajp10304/ajp10304.c

```diff
@@ -0,0 +1,138 @@
+#include "ajp10304.h"
+
+void persistant_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+   switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistant_default_layer_set(1UL<<_QWERTY);
+      }
+      return false;
+    case LOWER:
+      if (record->event.pressed) {
+        layer_on(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+    case RAISE:
+      if (record->event.pressed) {
+        layer_on(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+    case MLWR:
+      if (record->event.pressed) {
+        layer_on(_LOWER);
+        layer_on(_MLWR);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_LOWER);
+        layer_off(_MLWR);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+    case MRSE:
+      if (record->event.pressed) {
+        layer_on(_RAISE);
+        layer_on(_MRSE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_RAISE);
+        layer_off(_MRSE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+    case MFNC:
+      if (record->event.pressed) {
+        layer_on(_FUNC);
+        layer_on(_MFNC);
+      } else {
+        layer_off(_FUNC);
+        layer_off(_MFNC);
+      }
+      return false;
+    case MFNC2:
+      if (record->event.pressed) {
+        layer_on(_FUNC2);
+        layer_on(_MFNC2);
+      } else {
+        layer_off(_FUNC2);
+        layer_off(_MFNC2);
+      }
+      return false;
+    }
+  return true;
+}
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t keycode, uint8_t opt) {
```

Comment:
This way of defining macros is deprecated.

Reply:
To clarify, could you move these macros to the `process_record_user()` style above?

Reply:
Yup, will get to this.

Reply:
Done.